### ISSUE TITLE
Review Windows Support Documentation

### ISSUE THREAD

**Issue by tanabi** [fuzzball-muck/fuzzball#576]
It's been determined that the windows binaries do, indeed, still work.  However, we should make sure our documentation is up to date.  Specifically we need to make sure the user has the VC redist installed for x86.

 https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads

It looks like the windows build lacks any top-level README file so perhaps we should put one in there with some how-to documentation.

**Comment by wyld-sw** [comment]
@foxbird Would you be able to determine what we've broken? We definitely appreciate the help.

**Comment by digitalcircuit** [comment]
Though I'm fairly inexperienced when it comes to Windows, I'm willing to try to help with fixes with regards to the continuous integration setup.  @foxbird, if you've got a working way to build Fuzzball locally, I'm happy to try to correct the CI to match.

(I imagine a lot has changed in the past years.  The project I originally took inspiration from, Quassel IRC, has wound up migrating Windows build systems as well.)

**Comment by foxbird** [comment]
TLDR; Everything works fine as expected. If there's a specific database being used, maybe there's a bug causing a crash. I'd be happy to look into that.

I took a look at appveyor and it says the latest build was successful with no issues. So, I thought perhaps that something strange was going on and grabbed the latest master and decided to build it. Here were the steps with the latest visual studio.

1. I haven't had openssl installed in a while, so I snagged that from here https://slproweb.com/products/Win32OpenSSL.html. Specifically, I wanted the 32 bit version (we could convert it to a 64 bit edition is we're all that keen on doing so). I installed the OpenSSL distribution into `Z:\OpenSSL_1.0.2u-x86` and told it to put the binaries in the `bin` folder and not my system directory (more on that later).
2. I have my git repo in `Z:\fuzzball\fuzzball`, so I made sure it was up to date.
3. I launched the `x86 native tools command prompt for VS 2019` from my start menu.
4. I changed into `Z:\fuzzball\fuzzball`.
5. I ran `nmake /f makefile.win /D "OPENSSLDIR=z:\OpenSSL_1.0.2u-x86"`. There were a few warnings generated (some incompatible types and differing const qualifiers), which I dutifully ignored for now.
6. I then copied an old `minimal.db` I had hanging around and put it in the data directory (bad idea apparently).
7. I changed into the `game` directory where it put my binaries.
8. I ran `fbmuck data/minimal.db data/minimal.out 8888`.
9. First, I got a `couldn't find ssleay` and `couldn't find libeay' messages. So I copied them into the game directory from the open ssl install (see above about the not installing into system directory). The appeveyor artifact has them in there already.
9. It blipped briefly and then quit. The logfile had an entry for loading minimal.db, then nothing. Maybe the database was bad.
10. I attached the debugger and sure enough, it wasn't liking the database. I swapped out for minimal.db (told you using the old one was a bad idea) in the `dbs` directory and tried again. Success!

So, overall, the build process seems fine. I checked appveyor's output and it seems identical to mine (except the VS version differences). So, I also snagged the builds from appveyor and dropped them on top of my locally built ones. I reran them, and it's all going peachy keen. No errors, no hex codes, nothing.

Now, just to be sure, I did try this out on a fresh windows machine. Keep in mind, appveyor isn't building a release, just an artifact of what was committed. It's quite likely that there's a VC distribution that needs to be installed before it works. So, I'm kind of expecting that to be the case when I run this.

So I copied over my game directory from appveyor with the good minimal.db to a brand new windows machine, hit the command prompt, and got a message about not being able to find the VC redist library (more generic dll not found goodness). Not totally surprised. I expected that as I mentioned above. If this machine weren't new and had some installed software, I'd probably already have it. But, as it is, I go and install it from: https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads. We want the VS2019 version, and because this is 32bit, we want x86. The readme markdown for windows indicates this and covers the 32bit part as well. After installing it, I reran the command and I was up and running. I needed to add a telnet client to the windows features, but it connected and let me browse around my dark room zero without any issue!

I'm also sure the scary 'we've never seen this program run on your machine before' message could trip some people up (I didn't get it on either box, but I also copied it in from other media not from the internet). It's unsigned, downloaded from the net, and windows defender has never scanned it before. So microsoft isn't going to want to run it without a fight. But, once I had 'approved' it, it was happy to run. No 'compatibility mode' needed. It's a native app. It's just not trusted. And trusting it is as easy as clicking that 'more options' and saying 'run anyway'. Also, keep in mind that this isn't a GUI app. It's a command line thing, so double-clicking the exe won't do much. If wanted to promote this to a full on windows service, with an app tray icon and such, that'd take a good bit of dev work. Probably not worth the effort over a good readme (which the markdown one covers).

Personally, I'd rather not remove the windows bits from the source. I'm happy to support and test (or even contribute) to simplifying the places where there is some, and isolating it into a single file. Then you don't have to see or think about it, unless the code isn't standard C, or if a special library is being used. I don't really think there is all that much of it. Of course, if you want to remove it, I'm happy to maintain my own fork of fuzzball and keep it relatively up to date. The burden isn't that great, and I did it in the past for years. I'd rather not have to, but to support those folks that want native windows builds (all 2 of them I'm sure), I'm happy to.

**Comment by tanabi** [comment]
@foxbird If it works and works in a documented way, I'm fine with keeping it around.  I just genuinely believed it was no longer working.  I'll review your notes here in more detail and try it again on my VM later.  My VM is a fresh Windows 10 install, up to date with patches.  Not sure why I got a hex code message and you didn't, but I didn't have openssl installed or VC redist so I'll go through your steps and see if I can get it to work.

**Comment by foxbird** [comment]
You will need the vcredist installed. That's a requirement. The openssl binary you don't need as appveyor has it already packaged with what was built (it happens that appveyor is using a `s` release and the windows binaries I linked to are for `u` release). It should be as simple as this:

1. Download the zip package from appveyor.
2. Extract it into say c:\game (or somewhere else without spaces)
3. Make sure to install the vcredist if you haven't.
4. Put in a known good minimal.db into the data directory.
5. Drop to a command prompt and change to your game directory.
6. `fbmuck data\minimal.db data\minimal.out 8888` and it should run.

You will probably quite a firewall warning about letting it open ports which you should allow. And you may get the dreaded 'this isn't signed, don't trust it' message. To get round that, click 'more options' (or maybe more info) in the bottom left, and then there should be a button that is 'Run anyway'. Both are typical windows popups that most windows users are kind of used to unfortunately.

If we wanted to solve the code signing thing, we could, but that'd require someone buying a code signing cert. Doable, but not exactly the kind of thing you do unless you're serious about running it.

**Comment by tanabi** [comment]
@foxbird @wyld-sw The problem was the VC redist -- USUALLY I have gotten a friendly error suggesting that I install the visual C redist, but I think the fact that I'm running a 64 bit OS and trying to load a 32 bit binary perhaps means it isn't detecting correctly.

No matter, when I installed the x86 vc redist the MUCK sprung to life.  So that was the problem.  I think we need to review our windows documentation, but, I rescend my recommendation to decommission windows.

**Comment by tanabi** [comment]
@digitalcircuit 's commit resolves this issue I believe.
